lib/dialects/abstract/query-generator.js

```diff
@@ -1484,48 +1483,68 @@ const QueryGenerator = {
   getQueryOrders(options, model, subQuery) {
     const mainQueryOrder = [];
     const subQueryOrder = [];
+    const validOrderOptions = [
+      'ASC',
+      'DESC',
+      'ASC NULLS LAST',
+      'DESC NULLS LAST',
+      'ASC NULLS FIRST',
+      'DESC NULLS FIRST',
+      'NULLS FIRST',
+      'NULLS LAST'
+    ];
+
+    const methodOrder = order => {
+      let result = false;
+      if (
+          order._isSequelizeMethod
+          && (
+            order instanceof Utils.Fn
+            || order instanceof Utils.Col
+            || order instanceof Utils.Literal
+          )
+      ){
+        result = true;
+      }
+      return result;
+    };
 
     const validateOrder = order => {
-      if (order instanceof Utils.Literal) return;
-
-      if (!_.includes([
-        'ASC',
-        'DESC',
-        'ASC NULLS LAST',
-        'DESC NULLS LAST',
-        'ASC NULLS FIRST',
-        'DESC NULLS FIRST',
-        'NULLS FIRST',
-        'NULLS LAST'
-      ], order.toUpperCase())) {
+      if (methodOrder(order)) {
+        return;
+      }
+
+      if (!Utils._.includes(validOrderOptions, order.toUpperCase())) {
         throw new Error(util.format('Order must be \'ASC\' or \'DESC\', \'%s\' given', order));
       }
     };
 
     if (Array.isArray(options.order)) {
-      for (const t of options.order) {
-        if (Array.isArray(t) && _.size(t) > 1) {
-          if ((typeof t[0] === 'function' && t[0].prototype instanceof Model) || (typeof t[0].model === 'function' && t[0].model.prototype instanceof Model)) {
-            if (typeof t[t.length - 2] === 'string') {
-              validateOrder(_.last(t));
+      for (const order of options.order) {
+        if (Array.isArray(order) && Utils._.size(order) > 1) {
+          if (Utils._.isFunction(order[0]) && order[0].prototype instanceof Model || Utils._.isFunction(order[0].model) && order[0].model.prototype instanceof Model) {
+            if (Utils._.isString(order[order.length - 2])) {
+              validateOrder(Utils._.last(order));
             }
           } else {
-            validateOrder(_.last(t));
+            validateOrder(Utils._.last(order));
           }
         }
 
-        if (subQuery && (Array.isArray(t) && !(typeof t[0] === 'function' && t[0].prototype instanceof Model) && !(t[0] && typeof t[0].model === 'function' && t[0].model.prototype instanceof Model))) {
-          subQueryOrder.push(this.quote(t, model, false, '->'));
+        if (subQuery && (Array.isArray(order) && !(Utils._.isFunction(order[0]) && order[0].prototype instanceof Model) && !(order[0] && Utils._.isFunction(order[0].model) && order[0].model.prototype instanceof Model))) {
```

Comment:
Why the change to lodash typechecking functions everywhere? `typeof` is way faster than lodash's cross-browser implementation.